f2/src/utils/validateDate.js

```diff
@@ -0,0 +1,56 @@
+//import isExists from 'date-fns/isExists'
+import isFuture from 'date-fns/isFuture'
+import getDaysInMonth from 'date-fns/getDaysInMonth'
+import { containsData } from './containsData'
+
+export const validateDate = (y, m, d) => {
+  const year = parseInt(y, 10)
+  const monthIndex = parseInt(m, 10) - 1 // 0 indexed months
+  const day = parseInt(d, 10)
+  const date = new Date(year, monthIndex, day)
+
+  let validate = []
+  let isYear, isMonth, isDay
+
+  const hasYear = containsData(year)
+  const hasMonth = containsData(monthIndex)
+  const hasDay = containsData(day)
+
+  // 1. Date must contain a day a month and a year
+  // 2. Date must be in the past
+  // 3. Field ...X is not valid
+
+  // day is between 1 and the month's last day.
+  // Requires a date to validate against.
+  // new Date with year and month to prevent date-forwarding (jan 32 == feb 1)
+  if (day > getDaysInMonth(new Date(year, monthIndex)) || day < 1)
+    isDay = 'notDay'
+
+  // month is between 0 and 11
+  if (monthIndex > 11 || monthIndex < 0) isMonth = 'notMonth'
+
+  // is not in the future, while all fields are valid
+  !isMonth && !isDay && !isYear && isFuture(date) && validate.push('isFuture')
+
+  //Year is over 1867 :eyes:
```

Comment:
this is a requirement?

Reply:
No, I just don't know why we are accepting dates as far a year 1000, or even year 0001. I don't think this would affect people really, just a troll-barrier-easter-egg. We can remove this condition if we want!

Reply:
Actually I feel it is good idea to set  the year 150 years from now. If report input the year before 1867. It probably they made mistake. who can live that long and there was no computer at that time. I also check UK'S antifraud web: https://reporting.actionfraud.police.uk/reporting-form  The effective year is after 1900. That's my opinion. I will post a message  to get other team members opinion